x/mongo/driver/topology/rtt_monitor.go

```diff
@@ -190,6 +195,28 @@ func min(samples []time.Duration, minSamples int) time.Duration {
 	return min
 }
 
+// percentile returns the specified percentile value of the slice of duration samples. Zero values
+// are not considered samples and are ignored. If no samples or fewer than minSamples are found
+// in the slice, percentile returns 0.
+func percentile(perc float64, samples []time.Duration, minSamples int) time.Duration {
+	// Convert Durations to float64s.
+	floatSamples := make([]float64, 0)
+	for _, sample := range samples {
+		if sample > 0 {
+			floatSamples = append(floatSamples, float64(sample))
+		}
+	}
+	if len(floatSamples) < minSamples {
+		return 0
+	}
+
+	p, err := stats.Percentile(floatSamples, perc)
+	if err != nil {
+		panic(fmt.Errorf("error calculating %f percentile RTT: %v", perc, err))
```

Comment:
Optional: If we ever get this panic, we really want to know what all inputs to `Percentile` are. Consider including the `floatSamples` in the panic output as well.

Optional: Consider including the package where the panic came from in case someone wraps the panic and loses the stack trace. E.g.
```go
panic(fmt.Errorf("x/mongo/driver/topology: error calculating %f percentile RTT: %v", perc, err))
```

Reply:
Great point; elaborated the `panic` message as described above and also included the `floatSamples` slice in the output (the error message from `stats.Percentile` can be a little vague).